components/brave_rewards/resources/tip/lib/interfaces.ts

```diff
@@ -0,0 +1,182 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+type EmptyMediaData = {
+  mediaType: 'none'
+}
+
+export type TwitterMediaData = {
+  mediaType: 'twitter'
+  twitterName: string
+  screenName: string
+  userId: string
+  tweetId: string
+  tweetTimestamp: number
+  tweetText: string
+}
+
+type RedditMediaData = {
+  mediaType: 'reddit'
+  userName: string
+  postText: string
+  postRelDate: string
+}
+
+type GithubMediaData = {
+  mediaType: 'github'
+  userName: string
+}
+
+export type MediaMetaData =
+  EmptyMediaData |
+  TwitterMediaData |
+  RedditMediaData |
+  GithubMediaData
+
+export type ShareTarget = 'twitter'
+
+export type EntryPoint =
+  'one-time' |
+  'set-monthly' |
+  'clear-monthly'
+
+export interface DialogArgs {
+  url: string
+  entryPoint: EntryPoint
+  publisherKey: string
+  mediaMetaData: MediaMetaData
+}
+
+export type TipKind = 'one-time' | 'monthly'
+
+export type PaymentKind = 'bat'
+
+export enum PublisherStatus {
+  NOT_VERIFIED = 0,
+  CONNECTED = 1,
+  VERIFIED = 2
+}
+
+export interface PublisherInfo {
+  publisherKey: string
+  name: string
+  title: string
+  description: string
+  background: string
+  logo: string
+  amounts: number[]
+  provider: string
+  links: Record<string, string>
+  status: PublisherStatus
+}
+
+export interface BalanceInfo {
+  total: number
+  wallets: Record<string, number>
+}
+
+export enum ExternalWalletStatus {
+  NOT_CONNECTED = 0,
+  CONNECTED = 1,
+  VERIFIED = 2,
+  DISCONNECTED_NOT_VERIFIED = 3,
+  DISCONNECTED_VERIFIED = 4
+}
+
+export interface ExternalWalletInfo {
+  token: string
+  address: string
+  status: ExternalWalletStatus
+  type: 'anonymous' | 'uphold'
+  verifyUrl: string
+  addUrl: string
+  withdrawUrl: string
+  userName: string
+  accountUrl: string
+  loginUrl: string
+}
+
+export interface RewardsParameters {
+  rate: number
+  tipChoices: number[]
+  monthlyTipChoices: number[]
+}
+
+export type StringKey =
+  'addFunds' |
+  'bap' |
+  'bapFunds' |
+  'bat' |
+  'batFunds' |
+  'cancel' |
+  'cancelConfirmationText' |
+  'cancelMonthlyContribution' |
+  'changeAmount' |
+  'confirmCancel' |
+  'contributionAmount' |
+  'contributionCanceled' |
+  'currentlySupporting' |
+  'currentMonthlyContribution' |
+  'doMonthly' |
+  'errorHasOccurred' |
+  'githubTipTitle' |
+  'githubTipTitleEmpty' |
+  'monthlyContributionSet' |
+  'monthlyText' |
+  'nextContributionDate' |
+  'notEnoughTokens' |
+  'notEnoughTokensLink' |
+  'on' |
+  'oneTimeTip' |
+  'oneTimeTipAmount' |
+  'points' |
+  'redditTipTitle' |
+  'redditTipTitleEmpty' |
+  'rewardsBannerText1' |
+  'sendDonation' |
+  'siteBannerConnectedText' |
+  'siteBannerNoticeNote' |
+  'siteBannerNoticeText' |
+  'sorryToSeeYouGo' |
+  'supportThisCreator' |
+  'termsOfService' |
+  'thanksForTheSupport' |
+  'tipHasBeenSent' |
+  'tipPostSubtitle' |
+  'tokens' |
+  'tweetAboutSupport' |
+  'tweetTipTitle' |
+  'tweetTipTitleEmpty' |
+  'unVerifiedTextMore' |
```

Comment:
Nit: Doesn't seem like the 'v' needs to be capitalized.

Reply:
Fixed.